src/components/App.js

```diff
@@ -1,14 +1,66 @@
 import React, { Component } from 'react'
+import beastData from '../beastdata.json';
 import Header from './Header.js';
 import Main from './Main.js';
 import Footer from './Footer.js';
+import SelectedBeast from './SelectedBeast.js';
+import InputGroup from 'react-bootstrap/InputGroup';
+import FormControl from 'react-bootstrap/FormControl';
 
 export default class App extends Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      modalDisplayed: false,
+      clickedBeast: {},
+      search: ""
+    }
+  }
+  updateClickedBeast = (beastObj) => {
+    this.setState({clickedBeast: beastObj});
+  };
+  showModal = () =>{
+    this.setState({modalDisplayed: true});
+  };
+  hideModal = () =>{
+    this.setState({modalDisplayed: false});
+  };
+  textChange = (e) => {
+    this.setState({search: e.target.value})
+    this.filterSearch();
+    console.log(this.state.search);
+  };
```

Comment:
Great work on the lab!

Consider adding one line of space between each method for easier readability  and one line of space after the closing `}` for your constructor for easier readability 👍